pkg/controllers/jobset_controller.go

```diff
@@ -576,10 +574,9 @@ func constructJob(js *jobset.JobSet, rjob *jobset.ReplicatedJob, jobIdx int) (*b
 	labelAndAnnotateObject(job, js, rjob, jobIdx)
 	labelAndAnnotateObject(&job.Spec.Template, js, rjob, jobIdx)
 
-	// If enableDNSHostnames is set, update job spec to set subdomain as
-	// job name (a headless service with same name as job will be created later).
-	if dnsHostnamesEnabled(rjob) {
-		job.Spec.Template.Spec.Subdomain = GenSubdomain(js, rjob)
+	// If enableDNSHostnames is set, a Subdomain should be set
+	if dnsHostnamesEnabled(js) && js.Spec.Network.Subdomain == "" {
```

Comment:
we need to do this check before trying to create the service (line 355), not here.